lib/plug/adapters/cowboy.ex

```diff
@@ -143,8 +145,11 @@ defmodule Plug.Adapters.Cowboy do
     acceptors = cowboy_options[:acceptors] || 100
     dispatch  = :cowboy_router.compile(cowboy_options[:dispatch])
     compress  = cowboy_options[:compress] || false
-    cowboy_options   = Keyword.drop(cowboy_options, @not_cowboy_options)
-    [ref, acceptors, cowboy_options, [env: [dispatch: dispatch], compress: compress]]
+    timeout_option    = if cowboy_options[:timeout] do [timeout: cowboy_options[:timeout]] else [] end
```

Comment:
It'd be better to use keyword-style if for one-liner: `if timeout = cowboy_options[:timeout], do: [timeout: timeout], else: []`.